parsec/class/info.c

```diff
@@ -311,10 +314,26 @@ void *parsec_info_get(parsec_info_object_array_t *oa, parsec_info_id_t iid)
     ie = parsec_info_lookup_by_iid(oa->infos, iid);
     if(NULL == ie->constructor)
         return ret;
+    if(NULL != oa->ctx_set) oa->ctx_set(oa->ctx_set_obj);
     nio = ie->constructor(oa->cons_obj, ie->cons_data);
     ret = parsec_info_test_and_set(oa, iid, nio, NULL);
     if(ret != nio && NULL != ie->destructor) {
         ie->destructor(nio, ie->des_data);
     }
     return ret;
 }
+
+void parsec_info_set_all(parsec_info_t *nfo, parsec_info_id_t iid)
```

Comment:
Suggest a different name:
```suggestion
void parsec_info_populate(parsec_info_t *nfo, parsec_info_id_t iid)
```

Reply:
Let me question the need for this function. It is only use in a single location, and based on what it is doing I don't see the need for a capability to populate the entire info set. Second issue with this populate capability, is that all memory allocation will be done in the context of the caller of the populate, instead of allowing the allocated objects to belong to the scope of the thread/stream that would use them.

In fact my main issue is with the entire info API. Because `parsec_info_set` nicely allows you to set a value for a key, but then `parsec_info_get` will magically allocate one for you even if you never set one. And there is no way to distinguish between the case where the value was properly set and the case where it was auto-generated.